### PR TITLE
This is hopefully the final cleanup of the Scene class.

### PR DESCRIPTION
**Pull request by moorepants**
- [x] Check to see if all viz examples run on Py 2 and 3.
- [x] There are no merge conflicts.
- [x] If there is a related issue, a reference to that issue is in the
  commit message.
- [x] Unit tests have been added for the bug. (Please reference the issue #
  in the unit test.)
- [x] The tests pass both locally (run `nosetests`) and on Travis CI.
- [x] The code follows PEP8 guidelines. (use a linter, e.g.
  [pylint](http://www.pylint.org), to check your code)
- [x] The bug fix is documented in the [Release
  Notes](https://github.com/pydy/pydy#release-notes).
- [x] The code is backwards compatible. (All public methods/classes must
  follow deprecation cycles.)
- [x] All reviewer comments have been addressed.

### CODE REVIEW

**Comment by moorepants** [comment]
@sahilshekhawat @tarzzz This works except for one thing. When you click `Rerun` in the mass_spring_damper example the `jQuery($load-simulation).click()` function does not seem to execute. I can't figure out what prevents that. Help needed and then this is hopefully ready.

**Comment by sahilshekhawat** [comment]
Working on it.

**Comment by sahilshekhawat** [comment]
I tried this PR and it is working. jQuery($load-simulation).click() is executing.

**Comment by moorepants** [comment]
When I change the mass to 100 and click rerun the animation should show the mass moving a further distance, but it doesn't. Also the console log for the `click()` function does not seem to get caught when you press the button. How do you know it is working?

**Comment by sahilshekhawat** [comment]
I changed mass to 100 and it is moving further distance. I am not sure why is it now working on your system.

**Comment by moorepants** [comment]
Let me try again. I'm using Firefox.

**Comment by moorepants** [comment]
When I press rerun the console does not show this PyDy Info line:

https://github.com/pydy/pydy/blob/master/pydy/viz/static/js/dyviz/dv.js#L64

Does it for you?

The mass distance definitely doesn't change for me.

**Comment by moorepants** [comment]
Can you try it on Firefox?

File: pydy/viz/static/js/dyviz/materials.js [hunk 1]
```diff
@@ -14,7 +14,8 @@ DynamicsVisualizer.Materials = Object.extend(DynamicsVisualizer, {
 
         var self = this;
 
-        self.texture_base_path = "static/textures/"
+        // TODO : This `pydy-resources` directory should not be hard coded here!
+        self.texture_base_path = "pydy-resources/textures/"
```
Comment (moorepants) [inline pydy/viz/static/js/dyviz/materials.js]: Sahil, how can we properly fix this? The textures need to be loaded relatively but this is a hard coded path.

**Comment by sahilshekhawat** [comment]
Oh! yes, i forgot to checkout branch after fetching,

**Comment by moorepants** [comment]
Here is a quick setup for checkout out a PR and testing: https://github.com/moorepants/dotfiles/blob/master/gitconfig#L20

You then just do `git pr 245`.

**Comment by sahilshekhawat** [comment]
I am just sleepy. :smile:

**Comment by moorepants** [comment]
You can go to bed and look at it in the morning.

**Comment by sahilshekhawat** [comment]
I found the issue. The code for the button with id "load-simulation" is not even there in index_ipython.html. You just have to replace that code and it will work.

File: pydy/viz/static/index_ipython.html [hunk 2]
```diff
@@ -37,9 +37,9 @@
     <div id="wrapper">
       <div id="container">
         <div id="sidebar-right" >
-          <div id="json-inp">
+          <div id="json-inp" style="display:none;">
+            <!--Used in Ipython notebook to set sceneFileURI to load in dv.js-->
             <input type="text" class="form-control" id="json-input" value="{load_url}">
-            <input type="button" class="btn btn-primary" value="Load Simulation" id="simulation-load">
```
Comment (sahilshekhawat) [inline pydy/viz/static/index_ipython.html]: Add back this line.

Comment (moorepants) [inline pydy/viz/static/index_ipython.html]: Thanks. I'll fix.

Comment (moorepants) [inline pydy/viz/static/index_ipython.html]: Sweet, that was it!

File: pydy/viz/static/js/dyviz/materials.js (hunk 1 above)
Comment (sahilshekhawat) [inline pydy/viz/static/js/dyviz/materials.js]: We can add a script tag at the end of html files e.g.
```
DynamicsVisualizer.Materials.texture_base_path = {pydy-resources-dir}
```
and can set it like we are doing in display_ipython
```
html = html.format(pydy-resources-dir=self.pydy_reousrces_dir)
```

Comment (moorepants) [inline pydy/viz/static/js/dyviz/materials.js]: Ok.

**Comment by moorepants** [comment]
Sahil and I have reviewed this back and forth enough that I think that is sufficient. I'm gonna merge.
